src/components/DepositWidget/index.tsx

```diff
@@ -60,9 +68,28 @@ const LinkWrapper = styled.div`
   }
 `
 
+const txOptionalParams: TxOptionalParams = {
+  onSentTransaction: (receipt: Receipt): void => {
+    const notification = <TxNotification txHash={receipt.transactionHash} />
+    if (notification) {
+      toast.info(notification)
+    } else {
+      console.error(`Failed to get notification for tx ${receipt.transactionHash}`)
+    }
+  },
+}
+
 const DepositWidget: React.FC = () => {
-  const { balances, error } = useTokenBalances()
+  const { userAddress } = useWalletConnection()
+  const { balances, setBalances, error } = useTokenBalances()
+
   const contractAddress = depositApi.getContractAddress()
+  const mounted = useRef(true)
+  useEffect(() => {
+    return function cleanUp(): void {
+      mounted.current = false
```

Comment:
Question: are we using this mounted-check logic often? I may have a proposal for a generic hook `useIfMounted` if we need it

Reply:
I've seen it used a lot, go ahead and make an issue for it I'd be interested in seeing it's implementation